### PR TITLE
Failing test for gh-13945

### PR DESCRIPTION
**Pull request by mdeinum**
Failing testcase to show the issue. When Spring AOP need method
arguments to build a cache key (but probably also on other annotations
that require method arguments) the AOP part will fail.

Creating a Mockito mock results in the loss of the debug information
needed to read the correct method parameter.

issue: gh-13945

### CODE REVIEW

File: spring-boot-project/spring-boot-test/src/test/java/org/springframework/boot/test/mock/mockito/SpyBeanWithAopProxyTests.java [hunk 1]
```diff
@@ -64,6 +65,18 @@ public void verifyShouldUseProxyTarget() throws Exception {
 		verify(this.dateService, times(1)).getDate(matchesAnyBoolean());
 	}
 
+	@Test
+	public void verifyShouldUseProxyTargetWithMethodArguments() throws Exception {
+		Long d1 = this.dateService.getRandomDate(1234L);
+		Thread.sleep(200);
+		Long d2 = this.dateService.getRandomDate(1234L);
+		assertThat(d1).isEqualTo(d2);
```
Comment (wilkinsona) [inline spring-boot-project/spring-boot-test/src/test/java/org/springframework/boot/test/mock/mockito/SpyBeanWithAopProxyTests.java]: This assertion passes which means that the caching using `#seed` is working

File: spring-boot-project/spring-boot-test/src/test/java/org/springframework/boot/test/mock/mockito/SpyBeanWithAopProxyTests.java [hunk 2]
```diff
@@ -64,6 +65,18 @@ public void verifyShouldUseProxyTarget() throws Exception {
...
+		verify(this.dateService, times(1)).getDate(false);
+		verify(this.dateService, times(1)).getDate(matchesFalse());
+		verify(this.dateService, times(1)).getDate(matchesAnyBoolean());
```
Comment (wilkinsona) [inline spring-boot-project/spring-boot-test/src/test/java/org/springframework/boot/test/mock/mockito/SpyBeanWithAopProxyTests.java]: These three verifications are verifying the wrong method. They should be checking that `getRandomDate(long)` has only been called once:

```java
verify(this.dateService, times(1)).getRandomDate(1234L);
```

With this change in place, the test passes.

**Comment by wilkinsona** [comment]
Thanks for the PR, @mdeinum. Unfortunately the provided test does not appear to reproduce the problem. I'll take a look at 
https://github.com/spring-projects/spring-boot-issues/pull/79 instead.

File: spring-boot-project/spring-boot-test/src/test/java/org/springframework/boot/test/mock/mockito/SpyBeanWithAopProxyTests.java (hunk 2 above)
Comment (mdeinum) [reply spring-boot-project/spring-boot-test/src/test/java/org/springframework/boot/test/mock/mockito/SpyBeanWithAopProxyTests.java]: Interestingly enough this failed at calling the actual method (so I didn't even get to the assertions). Will fix the assertions and see what happens.

**Pull request closed without merging**